code/modules/clothing/modular_armor/modular.dm

```diff
@@ -367,10 +349,9 @@
 	///Current varient selected.
 	var/current_variant
 
-/obj/item/clothing/head/modular/Initialize(mapload)
+/obj/item/clothing/head/modular/Initialize()
 	. = ..()
-	AddComponent(/datum/component/attachment_handler, attachments_by_slot, attachments_allowed, attachment_offsets, starting_attachments, null, null, null, attachment_overlays)
-	update_icon()
+	update_icon() //Update for greyscale.
```

Comment:
update icon doesnt call greyscale or am i missing something

Reply:
Without the update_icon(), the thing was initializing invisible.